sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/implementation/EncryptionProcessor.java

```diff
@@ -627,6 +663,20 @@ public static JsonNode toJsonNode(byte[] serializedBytes, TypeMarker typeMarker)
         throw BridgeInternal.createCosmosException(0, "Invalid or Unsupported Data Type Passed " + typeMarker);
     }
 
+    private String convertToBase64UriSafeString(byte[] bytesToProcess) {
+        String base64String = new String(Base64.getDecoder().decode(bytesToProcess));
+
+        // Base 64 Encoding with URL and Filename Safe Alphabet  https://datatracker.ietf.org/doc/html/rfc4648#section-5
+        // https://docs.microsoft.com/en-us/azure/cosmos-db/concepts-limits#per-item-limits, due to base64 conversion and encryption
+        // the permissible size of the property will further reduce.
+        return base64String.replaceAll("/","_").replaceAll("\\+","-");
+    }
+
+    private byte[] convertFromBase64UriSafeString(String base64UriSafeString) {
+        base64UriSafeString.replaceAll("_", "/").replaceAll("-", "\\+");
```

Comment:
Using a StringBuilder initialized with the original string will be more efficient here (and is what .Net did)
Your option allocates two new strings - while string builder would only create the string builder instance - the actual string doesn't need to be copied multiple times.

Reply:
StringBuilder in java doesn't have a similar replace method as it does in .Net. In java StringBuilder.replace works differently.

For example:
StringBuilder sb1 = new StringBuilder("HelloWorld");
StringBuilder sb2 = sb1.replace(5,10, "One");
Result: "HelloOne"

I created a utility method using StringBuilder to replace the characters.